### ISSUE TITLE
Proposal for new a11y check to disallow use of the autofocus attribute

### ISSUE THREAD

**Issue by judithhinlung** [ember-template-lint/ember-template-lint#2166]
The autofocus attribute is a global attribute that indicates an element should be focused on page load. Autofocus reduces accessibility by moving users to an element without warning and context. Screen reader users may be disoriented by the unexpected navigation, and users of  small devices could miss important information preceding the focused element.

This proposes the authoring of a new rule that disallows use of the autofocus attribute. This is essentially a port of JSXs A11y [no-autofocus](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-autofocus.md) rule.

Reference: [WCAG SC 2.4.3 Focus Order](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html)
